monkey/monkey_island/cc/services/attack/technique_reports/T1188.py

```diff
@@ -0,0 +1,35 @@
+from monkey_island.cc.services.attack.technique_reports import AttackTechnique
+from monkey_island.cc.models.monkey import Monkey
+from common.utils.attack_utils import ScanStatus
+
+__author__ = "VakarisZ"
+
+
+class T1188(AttackTechnique):
+
+    tech_id = "T1188"
+    unscanned_msg = "Monkey didn't use multi-hop proxy."
+    scanned_msg = ""
+    used_msg = "Monkey used multi-hop proxy."
+
+    @staticmethod
+    def get_report_data():
```

Comment:
On one hand, this can be done much more efficiently, if it's done during the creation of the monkey.
Whenever a new monkey is created, it'll maintain its proxy depth and initial tunneling monkey.

On the other hand, long tunnels aren't common, and in that case, this code takes O(n) time, so it's fine. Also, I'm unsure if it's worth spamming the monkey mongo object with these 2 additional fields.

@ShayNehmad - Up to you

Reply:
First of all - it's not O(n). Worst case is a "chain" of monkeys and in that case:
![image](https://user-images.githubusercontent.com/48879847/62308201-aa2d6180-b48d-11e9-8f4c-41712fb34a80.png)

Anyway, I think this it's better to move it to pre-processing like you suggested. No reason to compute this on demand, we know the report generation is slow, and then it will by O(1), and in my opinion - more readable and testable as well. 

I think adding fields to the monkey model is OK as long as they are named carefully. We will probably want to separate the Monkey model to sub-components in future, when our use of the new model sub-system matures, but for now it's now an issue.

Reply:
Shay is correct. Easiest way to figure this out is to count the nested loops: here we have a loop for each monkey, and then for each tunnel of that monkey, thus it's n^2. From my experience n^3 is when you start to "feel" the downtime, but that's in data processing applications, not when you have a 100 entries. That being said, we only need the `source_monkey` and `proxy_depth` fields added to the monkey. So the way I see it it's only a matter of whether we will use these fields somewhere else and if it's worth our time. I'm leaning towards "yes", but I'll leave it up to you to decide.